tools/Nethermind.Tools.Kute/Program.cs

```diff
@@ -15,73 +14,92 @@
 using Nethermind.Tools.Kute.ResponseTracer;
 using Nethermind.Tools.Kute.SecretProvider;
 using Nethermind.Tools.Kute.SystemClock;
+using System.CommandLine;
 
 namespace Nethermind.Tools.Kute;
 
 static class Program
 {
-    public static async Task Main(string[] args)
+    public static async Task<int> Main(string[] args)
     {
-        await Parser.Default.ParseArguments<Config>(args).WithParsedAsync(async config =>
+        CliRootCommand rootCommand =
+        [
+            Config.MessagesFilePath,
+            Config.HostAddress,
+            Config.JwtSecretFilePath,
+            Config.AuthTtl,
+            Config.DryRun,
+            Config.ShowProgress,
+            Config.MetricsOutputFormatter,
+            Config.MethodFilters,
+            Config.ResponsesTraceFile,
+            Config.RequestsPerSecond,
+            Config.UnwrapBatch
+        ];
+        rootCommand.SetAction((parseResult, cancellationToken) =>
         {
-            IServiceProvider serviceProvider = BuildServiceProvider(config);
+            IServiceProvider serviceProvider = BuildServiceProvider(parseResult);
             Application app = serviceProvider.GetService<Application>()!;
 
-            await app.Run();
+            return app.Run();
         });
+
+        CliConfiguration cli = new(rootCommand);
+
+        return await cli.InvokeAsync(args);
     }
 
-    static IServiceProvider BuildServiceProvider(Config config)
+    private static IServiceProvider BuildServiceProvider(ParseResult parseResult)
     {
+        bool dryRun = parseResult.GetValue(Config.DryRun);
+        bool unwrapBatch = parseResult.GetValue(Config.UnwrapBatch);
+        string? responsesTraceFile = parseResult.GetValue(Config.ResponsesTraceFile);
```

Comment:
I think we can inline these variables, no need to have them defined here.

Reply:
They are being used in multiple places, so it fetches their values once.